### PR TITLE
Fi improve ts assign tm

### PR DESCRIPTION
**Pull request by fede-pe**
This PR includes:

- Enhancement of tilt series assign alignment protocol. Now we make the intersection of the two sets plus allow the combination of alignments.

### CODE REVIEW

File: tomo/protocols/protocol_assignTransformationTS.py [hunk 1]
```diff
@@ -48,50 +55,130 @@ def _defineParams(self, form):
...
         form.addParam('setTMSetOfTiltSeries',
                       params.PointerParam,
                       pointerClass='SetOfTiltSeries',
                       important=True,
                       help='Set of tilt-series on which transformation matrices will be assigned.',
-                      label='Tilt-series WITHOUT alignment')
+                      label='Tilt-series to assign the alignment to')
+
+        form.addParam('combineAlignment',
+                      params.EnumParam,
```
Comment (pconesa) [inline tomo/protocols/protocol_assignTransformationTS.py:68]: For this there are boolean params that are more convenient

File: tomo/protocols/protocol_assignTransformationTS.py [hunk 2]
```diff
@@ -48,50 +55,130 @@ def _defineParams(self, form):
...
+                      choices=['Yes', 'No'],
+                      default=0,
+                      label='Combine alignments',
+                      important=True,
+                      display=params.EnumParam.DISPLAY_HLIST,
+                      help='If the tilt-series to assign the alignment has a previous alignment'
+                           'this will be combined with the ')
```
Comment (pconesa) [inline tomo/protocols/protocol_assignTransformationTS.py:75]: with the ..? suspense .. wait for the next release.. ;-)

**Review by pconesa — CHANGES_REQUESTED** [review]
Just some minor changes.

File: tomo/protocols/protocol_assignTransformationTS.py (hunk 1 above)
Comment (fede-pe) [reply tomo/protocols/protocol_assignTransformationTS.py]: Fixed

File: tomo/protocols/protocol_assignTransformationTS.py (hunk 2 above)
Comment (fede-pe) [reply tomo/protocols/protocol_assignTransformationTS.py]: I was planning a premium feature... but you have a point :)

File: tomo/protocols/protocol_assignTransformationTS.py [hunk 3]
```diff
@@ -107,29 +194,27 @@ def updateTM(self, transform):
...
+        print(matrix)
+
         return transform
 
     # --------------------------- INFO functions ----------------------------
     def _validate(self):
         validateMsgs = []
 
-        for tsGetTM, tsSetTM in zip(self.getTMSetOfTiltSeries.get(), self.setTMSetOfTiltSeries.get()):
+        for tsGetTM in self.getTMSetOfTiltSeries.get():
```
[the remark this answers is not in the document]
Comment (fede-pe) [reply tomo/protocols/protocol_assignTransformationTS.py:208]: Fixed!!

**Review by pconesa — APPROVED** [review]

File: tomo/protocols/protocol_assignTransformationTS.py [hunk 4]
```diff
@@ -69,8 +69,8 @@ def _defineParams(self, form):
                       label='Combine alignments',
                       important=True,
                       display=params.EnumParam.DISPLAY_HLIST,
-                      help='If the tilt-series to assign the alignment has a previous alignment'
-                           'this will be combined with the ')
+                      help='If this option is selected and the the tilt-series to assign the alignment has a previous '
+                           'alignment, both precious and new alignment are combined.')
```
Comment (azazellochg) [inline tomo/protocols/protocol_assignTransformationTS.py:73]: precios : )

**Pull request merged**